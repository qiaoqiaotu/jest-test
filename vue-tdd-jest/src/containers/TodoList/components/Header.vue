<template>
  <div class="header">
    <div class="header-content">
      TodoList
      <input class="header-input"
             data-test="input"
             placeholder="请输入待办事项..."
             v-model="inputValue"
             @keyup.enter="addTodoItem" />
    </div>
  </div>
</template>

<script>
export default {
  name: 'Header',
  props: {
  },
  data () {
    return {
      inputValue: ''
    }
  },
  methods: {
    addTodoItem () {
      if (!this.inputValue) return
      this.$emit('add', this.inputValue)
      this.inputValue = ''
    }
  }
}
</script>

<style scoped lang="stylus">
.header
  background #666
  line-height 60px
  .header-content
    width 600px
    margin 0 auto
    color #FFF
    font-size 20px
    .header-input
      float right
      width 360px
      line-height 24px
      margin-top 16px
      outline none
      color #333
      text-indent 10px
</style>
