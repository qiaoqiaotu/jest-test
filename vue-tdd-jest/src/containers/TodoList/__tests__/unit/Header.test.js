import { shallowMount } from "@vue/test-utils";
import Header from "../../components/Header.vue";
import { findTestWrapper } from "../../../../utils/testUtils";
const wrapper = shallowMount(Header); // Header 组件
const input = findTestWrapper(wrapper, "input"); // input 框
const data = wrapper.vm.$data; // data

it("Header 样式发生改变时，提示", () => {
  expect(wrapper).toMatchSnapshot();
});
it("Header 包含一个 input 框", () => {
  expect(input.exists()).toBe(true);
});
it("Header 中的 input 发生变化，数据应该接着变化", () => {
  input.setValue("qiaoling");
  expect(data.inputValue).toBe("qiaoling");
});
it("Header 中的 input 输入回车，无内容时，无反应", () => {
  input.setValue("");
  input.trigger("keyup.enter");
  expect(wrapper.emitted().add).toBeFalsy();
});
it("Header 中的 input 输入回车，有内容时，有反应，同时清空 inputValue", () => {
  input.setValue("qiaoling");
  input.trigger("keyup.enter");
  expect(wrapper.emitted().add).toBeTruthy();
  expect(data.inputValue).toBe("");
});
