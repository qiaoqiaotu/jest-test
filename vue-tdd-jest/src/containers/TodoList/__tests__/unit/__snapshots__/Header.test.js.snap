// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 样式发生改变时，提示 1`] = `
<div class="header">
  <div class="header-content">
    TodoList
    <input data-test="input" placeholder="请输入待办事项..." class="header-input">
  </div>
</div>
`;

exports[`Header 组件 样式发生改变时，需要做提示 1`] = `
<div class="header">
  <div class="header-content">
    TodoList
    <input data-test="input" placeholder="请输入待办事项..." class="header-input">
  </div>
</div>
`;
